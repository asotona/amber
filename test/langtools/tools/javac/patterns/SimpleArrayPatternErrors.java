/**
 * @test nodynamiccopyright/
 * @summary Verify errors related to array patterns.
 * @compile/fail/ref=SimpleArrayPatternErrors.out -XDrawDiagnostics -XDshould-stop.at=FLOW --enable-preview -source ${jdk.version} SimpleArrayPatternErrors.java
 */
public class SimpleArrayPatternErrors {
    private boolean test(Object o) {
        return o instanceof {var s1} &&
               o instanceof String {var s2} &&
               o instanceof String[] {var s3, ..., var s4} &&
               o instanceof R({var s5}) &&
               o instanceof R(String {var s6}) &&
               o instanceof R(String[] {var s7, ..., var s8});
    }
    record R(String[] content) {}
}
