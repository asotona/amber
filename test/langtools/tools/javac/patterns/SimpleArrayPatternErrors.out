SimpleArrayPatternErrors.java:8:29: compiler.err.illegal.start.of.type
SimpleArrayPatternErrors.java:10:50: compiler.err.expected: '}'
SimpleArrayPatternErrors.java:13:52: compiler.err.expected: '}'
SimpleArrayPatternErrors.java:9:18: compiler.err.array.pattern.not.array: java.lang.String
SimpleArrayPatternErrors.java:12:31: compiler.err.array.pattern.not.array: java.lang.String
- compiler.note.preview.filename: SimpleArrayPatternErrors.java
- compiler.note.preview.recompile
5 errors
